### PR TITLE
Enable leader election in csi components

### PR DESCRIPTION
**Pull request by bcho**
- Settings are adapted from kubernetes-sigs/azuredisk-csi-driver#224
- Refactored chart image references

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```

```

### CODE REVIEW

File: deploy/csi-blobfuse-controller.yaml [hunk 1]
```diff
@@ -20,11 +20,13 @@ spec:
       priorityClassName: system-cluster-critical
       containers:
         - name: csi-provisioner
-          image: quay.io/k8scsi/csi-provisioner:v1.0.1
+          image: quay.io/k8scsi/csi-provisioner:v1.4.0
```
Comment (bcho) [inline deploy/csi-blobfuse-controller.yaml]: I updated these images to use the `leases` lock, is that safe to upgrade them? @andyzhangx

**Comment by andyzhangx** [comment]
@bcho e2e CI test is ready, could you rebase? Thanks.

**Comment by bcho** [comment]
> 
> 
> @bcho e2e CI test is ready, could you rebase? Thanks.

Sure, I will have a look later today

**Pull request merged**